source/portable/transport_interface.h

```diff
@@ -82,7 +82,13 @@
  * @snippet this define_transportrecv
  * <br>
  * This function is expected to populate a buffer, with bytes received from the
- * transport, and return the number of bytes placed in the buffer.
+ * transport, and return the number of bytes placed in the buffer. However, if a
+ * socket error occurs while trying to receive a message, then a negative error
```

Comment:
Let's not say socket error, since underlying transport could be using something other than socket.